shared/src/vyxal/MiscHelpers.scala

```diff
@@ -186,6 +187,9 @@ object MiscHelpers:
     vyPrint(x)
     vyPrint("\n")
 
+  def scanl(iterable: VList, function: VFun)(using ctx: Context): VList =
+    VList.from(ListHelpers.prefixes(iterable).map(reduce(_, function)))
```

Comment:
```suggestion
    if iterable.isEmpty then iterable
    else
      iterable.tail.scanLeft(iterable.head)(function(_, _))
```
Doing `prefixes` and reducing each prefix separately will be inefficient. If it turns out `scanLeft` on `VList`s requires evaluating them, which I don't think it does, but if that is the case, we can always modify `scanLeft` to be lazier